scx.cscheds: switch to make packaging

JohnRTitor opened the pull request:
Upstream deprecated meson builds.

This also switches libbpf and bpftool dependencies to nixpkgs provided ones. Please report if this causes any issues.




## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

JohnRTitor committed:
scx.cscheds: switch to make packaging

Upstream deprecated meson builds.

This also switches libbpf and bpftool dependencies to nixpkgs provided ones.
Please report if this causes any issues.

Gliczy reviewed (commented):
Thanks for tackling this!

We can drop `lib`, `jq` and `protobuf` in `scx_cscheds.nix`, since they are no longer used. Same thing about `finalAttrs`.

Both the rust and c scheds build fine, but `scx.full` fails to build. This should fix it:
```diff
# scx_full.nix
- cp ${scx.rustscheds}/bin/* ${placeholder "bin"}/bin/
+ cp ${scx.rustscheds}/bin/* $out/bin/
```

JohnRTitor committed:
scx: remove need for custom update script

nix-shell maintainers/scripts/update.nix --argstr package scx.rustscheds

Should be ran from now on

JohnRTitor committed:
scx: update homepage link
